python/scope.py

```diff
@@ -1662,7 +1664,9 @@ def type_check_expr(self, expr):
     """
 
     def type_check_command(self, tree):
-        if tree.data == "if" or tree.data == "ifelse":
+        if tree.data == "main_instruction" or tree.data == "last_instruction":
+            tree = tree.children[0]
```

Comment:
Another way is, in the syntax grammar, to do `?main-instruction` so it doesn't create a new Token